community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/schema/LuceneDocumentStructure.java

```diff
@@ -150,6 +150,14 @@ public static Query newWildCardStringQuery( String searchFor )
         return new WildcardQuery( term );
     }
 
+    public static Query newSuffixStringQuery( String suffix )
+    {
+        String searchTerm = QueryParser.escape( suffix );
+        Term term = new Term( ValueEncoding.String.key(), "*" + searchTerm );
```

Comment:
Since such kind of query will be terribly slow, did you consider possibility of using prefix search of reversed string instead?
Also please add benchmark for newly introduced query.